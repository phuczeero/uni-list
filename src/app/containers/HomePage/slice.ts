import { PayloadAction } from '@reduxjs/toolkit';
import { University } from 'types/University';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the HomePage container
export const initialState: ContainerState = {
  searchTerm: '',
  universities: [],
  loading: false,
  error: null,
};

const homePageSlice = createSlice({
  name: 'homePage',
  initialState,
  reducers: {
    loadUniversities(state, action: PayloadAction<string>) {
      state.searchTerm = action.payload;
      state.loading = true;
    },
    loadSuccess(state, action: PayloadAction<University[]>) {
      state.loading = false;
      state.universities = action.payload;
    },
    loadError(state, action: PayloadAction<Error>) {
      state.error = action.payload;
      state.loading = false;
    },
    saveToJSON() {},
  },
});

export const {
  actions: homePageActions,
  reducer,
  name: sliceKey,
} = homePageSlice;
