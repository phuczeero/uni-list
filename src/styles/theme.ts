const fonts = {
  body:
    '-apple-system, BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji"',
};
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 72];
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];
const radii = {
  default: 6,
  circle: 99999,
};
const colors = {
  text: '#212529',
  muted: '#dee2e6',
  primary: '#007bff',
  danger: '#dc3545',
  blues: {
    light: '#dcedff',
    dark: '#004896',
  },
};
const sizes = {
  navbar: 64,
  spinner: 32,
};
const shadows = {
  card: '0 0 4px rgba(0, 0, 0, .125)',
};
const breakpoints = ['576px', '768px', '992px', '1200px'];

export const theme = {
  fonts,
  fontSizes,
  space,
  radii,
  colors,
  sizes,
  shadows,
  breakpoints,
};

export type ThemeType = typeof theme;
