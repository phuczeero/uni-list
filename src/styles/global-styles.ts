import { createGlobalStyle } from 'styled-components';
import { ThemeType } from './theme';

export const GlobalStyle = createGlobalStyle<{ theme: ThemeType }>`
  * {
    font-family: ${p => p.theme.fonts.body};
    color: ${p => p.theme.colors.text}
  }
`;
