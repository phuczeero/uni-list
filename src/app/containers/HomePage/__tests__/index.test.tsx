import * as React from 'react';
import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/macro';
import { HelmetProvider } from 'react-helmet-async';
import { HomePage } from '..';
import { configureAppStore } from 'store/configureStore';
import { homePageActions, initialState } from '../slice';
import { theme } from 'styles/theme';

function* mockHomePageSaga() {}

jest.mock('../saga', () => ({
  homePageSaga: mockHomePageSaga,
}));

const renderHomePage = (store: Store) =>
  render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <HelmetProvider>
          <HomePage />
        </HelmetProvider>
      </ThemeProvider>
    </Provider>,
  );

describe('<HomePage />', () => {
  let store: ReturnType<typeof configureAppStore>;
  let component: ReturnType<typeof renderHomePage>;

  beforeEach(() => {
    store = configureAppStore();
    component = renderHomePage(store);
    expect(store.getState().homePage).toEqual(initialState);
  });
  afterEach(() => {
    component.unmount();
  });

  it('should display welcome message when searchTerm is empty', () => {
    expect(component.container.firstChild).toMatchSnapshot();
  });

  it('should display university list', () => {
    const searchTerm = 'test';
    store.dispatch(homePageActions.loadUniversities(searchTerm));
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
