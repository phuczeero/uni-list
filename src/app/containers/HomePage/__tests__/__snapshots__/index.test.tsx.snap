// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should display university list 1`] = `
.c0 {
  max-width: 100%;
  margin: auto;
  padding: 16px;
}

.c5 {
  outline: none;
  padding: 8px;
  border-radius: 6px;
  border: 2px solid #dee2e6;
}

.c3 {
  position: absolute;
  top: 10px;
  left: 10px;
}

.c2 {
  position: relative;
}

.c2 .c4 {
  padding-left: 32px;
}

.c7 {
  -webkit-animation: cqkDSr 2.625s linear infinite;
  animation: cqkDSr 2.625s linear infinite;
  height: 32px;
  width: 32px;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c8 {
  -webkit-animation: caVMmK 1.5s ease-in-out infinite;
  animation: caVMmK 1.5s ease-in-out infinite;
  stroke: #007bff;
  stroke-linecap: round;
}

.c6 {
  text-align: center;
  padding: 32px 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:576px) {
  .c0 {
    max-width: 576px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 768px;
  }
}

@media (min-width:992px) {
  .c0 {
    max-width: 992px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1200px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <form>
      <div
        class="c2"
      >
        <svg
          class="c3"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <input
          class="c4 c5"
          placeholder="Search universities"
        />
      </div>
    </form>
  </div>
  <div
    class="c6"
  >
    <svg
      class="c7"
      viewBox="-24 -24 48 48"
    >
      <circle
        class="c8"
        cx="0"
        cy="0"
        fill="none"
        r="20"
        stroke-width="4"
      />
    </svg>
  </div>
</div>
`;

exports[`<HomePage /> should display welcome message when searchTerm is empty 1`] = `
.c0 {
  max-width: 100%;
  margin: auto;
  padding: 16px;
}

.c5 {
  outline: none;
  padding: 8px;
  border-radius: 6px;
  border: 2px solid #dee2e6;
}

.c3 {
  position: absolute;
  top: 10px;
  left: 10px;
}

.c2 {
  position: relative;
}

.c2 .c4 {
  padding-left: 32px;
}

.c6 {
  padding: 16px;
  box-shadow: 0 0 4px rgba(0,0,0,.125);
}

.c7 {
  font-size: 32px;
  color: #dee2e6;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:576px) {
  .c0 {
    max-width: 576px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 768px;
  }
}

@media (min-width:992px) {
  .c0 {
    max-width: 992px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1200px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <form>
      <div
        class="c2"
      >
        <svg
          class="c3"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <input
          class="c4 c5"
          placeholder="Search universities"
        />
      </div>
    </form>
  </div>
  <div>
    <h2>
      Welcome to university list app.
    </h2>
    <div
      class="c6"
    >
      <div>
        <div
          class="c7"
        >
          To find your universities by name, enter the keyword to the search box above
        </div>
      </div>
    </div>
  </div>
</div>
`;
